File: Java/프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
class Solution {
    public int[] solution(String s) {
        int[] answer = new int[s.length()];
        for(int i=0;i<s.length();i++){
            answer[i]=f(s, i);
        }
        
        return answer;
    }
    int f(String s, int idx){
        for(int i = idx-1; i>-1;i--){
            
            if(s.charAt(i)==s.charAt(idx)){
                return idx-i;
            }
        }
        return -1;
    }
}

File: Kotlin/프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.kt
class Solution {
    fun solution(s: String): IntArray {
        var answer: IntArray = intArrayOf()
        var hm : HashMap<Char, Int> = HashMap()
        return s.mapIndexed{ index, v ->
            if(!hm.containsKey(v)) {
                -1
            } else {
                index-hm[v]!!
            }.also{
                hm[v] = index
            }
        }.toIntArray()
    }
}